src/lib/autoinstall/dialogs/question.rb

```diff
@@ -0,0 +1,179 @@
+    # Bear in mind that the timeout will not be respected when showing only
+    # the 'Abort' button.
+    #
+    # This dialog could be extended in the future in order to support other
+    # AutoYaST interaction which are not covered in the {Yast::Report} module.
+    class Question < UI::Dialog
+
+      # @return [String] Dialog's content
+      attr_reader :content
+
+      # Constructor
+      #
+      # @param content     [String]  Dialog's content
+      # @param timeout     [Integer] Countdown (in seconds); 0 means no timeout.
+      # @param buttons_set [Symbol]  Buttons set (:abort, :question)
+      def initialize(content, timeout: 10, buttons_set: :question)
+        @content = content
+        @remaining_time = timeout
+        @timed = timeout > 0
+        @buttons_set = buttons_set
+      end
+
+      # Return YaST terms for dialog's content
+      #
+      # @return [Yast::Term] Dialog's content
+      # @see UI::Dialog#dialog_content
+      def dialog_content
+        HBox(
+          VSpacing(20),
+          VBox(
+            HSpacing(70),
+            Left(Heading("Partitioning issues")),
+            VSpacing(1),
+            RichText(content),
+            timed? ? ReplacePoint(Id(:counter_replace), counter) : Empty(),
+            ButtonBox(*buttons)
+          )
+        )
+      end
+
+      # 'Continue' button handler
+      #
+      # When the 'Continue' button is pressed, the dialog will return the {:ok} value.
+      def ok_handler
+        finish_dialog(:ok)
+      end
+
+      # 'Abort' button handler
+      #
+      # When the 'Abort' button is pressed, the dialog will return the {:abort} value.
+      def abort_handler
+        finish_dialog(:abort)
+      end
+
+      # 'Stop' button handler
+      #
+      # When the 'Stop' button is pressed, the countdown will stop.
+      def stop_handler
+        @remaining_time = nil
+        Yast::UI.ChangeWidget(Id(:stop), :Enabled, false)
+      end
+
+      def timeout_handler
+        return finish_dialog(:ok) if @remaining_time.zero?
+        @remaining_time -= 1
+        Yast::UI.ReplaceWidget(Id(:counter_replace), counter)
+      end
+
+    private
+      # @return [Integer] Timeout
+      attr_reader :timeout
+
+      # @return [Symbol] Buttons set (:abort, :question)
+      attr_reader :buttons_set
+
+      # Determine whether it is a timed dialog or not
+      #
+      # @return [Boolean] True if the timeout is running; false otherwise
+      def timed?
+        @timed
+      end
+
+      # Return dialog buttons
+      #
+      # @see question_buttons
+      # @see abort_buttons
+      def buttons
+        send("#{buttons_set}_buttons")
+      end
+
+      # Return buttons to show when the user should be asked about what to do
+      #
+      # @return [Array<Yast::Term>]
+      def question_buttons
+        set = [
+          PushButton(Id(:ok), Opt(:okButton, :key_F9, :default), Yast::Label.ContinueButton),
```

Comment:
`F9` is usually used for `:cancel` or `:abort` actions, for `:ok` use the `F10` shortcut.